web/src/routes/OrganizationsList/Organization/Tabs/Settings/AutoPruning.tsx

```diff
@@ -0,0 +1,359 @@
+        variant: AlertVariant.Failure,
+        message: errorDeletePolicyDetails.toString(),
+      });
+    }
+  }, [errorDeletePolicy]);
+
+  const onSave = (e) => {
+    e.preventDefault();
+    let value = null;
+    switch (method) {
+      case AutoPruneMethod.TAG_NUMBER:
+        value = tagCount;
+        break;
+      case AutoPruneMethod.TAG_CREATION_DATE:
+        value = `${String(tagCreationDateValue)}${tagCreationDateUnit}`;
+        break;
+      case AutoPruneMethod.NONE:
+        // Delete the policy is done by setting the method to none
+        if (!isNullOrUndefined(uuid)) {
+          deletePolicy(uuid);
+        }
+        return;
+      default:
+        // Reaching here indicates programming error, component should always be aware of valid methods
+        return;
+    }
+    if (isNullOrUndefined(uuid)) {
+      createPolicy({method: method, value: value});
+    } else {
+      updatePolicy({uuid: uuid, method: method, value: value});
+    }
+  };
+
+  if (isLoading) {
+    return <Spinner />;
+  }
+
+  if (!isNullOrUndefined(error)) {
+    return <RequestError message={error.toString()} />;
+  }
+
+  return (
+    <>
+      <Title headingLevel="h2" style={{paddingBottom: '.5em'}}>
+        Auto Pruning Policies
+      </Title>
+      <p style={{paddingBottom: '1em'}}>
+        Auto-pruning policies automatically delete tags across all repositories
+        within this organization by a given method.
+      </p>
+      <Form id="autpruning-form" maxWidth="70%">
+        <FormGroup
+          isInline
+          label="Prune Policy - select a method to prune tags"
+          fieldId="method"
+          isRequired
+        >
+          <FormSelect
+            placeholder=""
+            aria-label="namespace-auto-prune-method"
+            data-testid="namespace-auto-prune-method"
+            value={method}
+            onChange={(_, val) => setMethod(val as AutoPruneMethod)}
+          >
+            <FormSelectOption
+              key={1}
+              value={AutoPruneMethod.NONE}
+              label="None"
+            />
+            <FormSelectOption
+              key={2}
+              value={AutoPruneMethod.TAG_NUMBER}
+              label="By number of tags"
+            />
+            <FormSelectOption
+              key={3}
+              value={AutoPruneMethod.TAG_CREATION_DATE}
+              label="By age of tags"
+            />
+          </FormSelect>
+          <FormHelperText>
+            <HelperText>
+              <HelperTextItem>The method used to prune tags.</HelperTextItem>
+            </HelperText>
+          </FormHelperText>
+        </FormGroup>
+        <Conditional if={method === AutoPruneMethod.TAG_NUMBER}>
+          <FormGroup label="The number of tags to keep." fieldId="" isRequired>
+            <NumberInput
+              value={tagCount}
+              onMinus={() => {
+                tagCount > 0 ? setTagCount(tagCount - 1) : setTagCount(0);
```

Comment:
If I save with number of tags as 0, I get a 400 error saying `Invalid Input`. I am guessing that [this](https://github.com/quay/quay/blob/master/data/model/autoprune.py#L46) line in the `valid_value` func is returning False. Mostly changing that line to `if value is None` should fix it.

Reply:
hmm would we want users to specify a repo with 0 tags?

Reply:
I instead added a check so a user can't input and empty value

Reply:
Yes, the check works. I don't think we might need a repo with 0 tags.